analytics/src/analytics/integrations/etldb/etldb.py

```diff
@@ -0,0 +1,41 @@
+"""Define EtlDb as an abstraction layer for database connections."""
+
+from enum import Enum
+
+from sqlalchemy import Connection
+
+from analytics.integrations import db
+
+
+class EtlDb:
+    """Encapsulate etl database connections."""
+
+    def __init__(self, effective: str | None = None) -> None:
+        """Construct instance."""
+        self._db_engine = db.get_db()
+        self.effective_date = effective
+        self.dateformat = "%Y-%m-%d"
+
+    def __del__(self) -> None:
+        """Destroy instance."""
+        self.disconnect()
+
+    def connection(self) -> Connection:
+        """Get a connection object from the db engine."""
+        return self._db_engine.connect()
```

Comment:
@DavidDudas-Intuitial when I try to run the `poetry run transform_and_load` with the full export I get the following error:

<img width="1420" alt="Screenshot 2024-11-01 at 10 54 36 AM" src="https://github.com/user-attachments/assets/bf8ed50b-9adf-4b9f-b105-169f939fb4d0">

It probably requires further digging, but I have a hunch that it might be stemming from this line, where you create a new connection since this is invoked for each record in a loop. 

I think a potential fix would be to use the top-level connection `self._db_engine` and either create a session or a connection that you pass in using dependency injection to be re-used throughout the loop, so that you're not spawning thousands of connections in the course of one run.